test/e2e/tests/network/multi-rpc.spec.ts

```diff
@@ -0,0 +1,461 @@
+
+        const arbitrumRpcUsed = await driver.findElement({
+          text: 'Arbitrum mainnet 2',
+          tag: 'button',
+        });
+
+        const existRpcUsed = arbitrumRpcUsed !== undefined;
+        assert.equal(existRpcUsed, true, 'Second Rpc is used');
+      },
+    );
+  });
+
+  it('should select rpc from settings @no-mmi', async function () {
+    async function mockRPCURLAndChainId(mockServer: Mockttp) {
+      return [
+        await mockServer
+          .forPost('https://responsive-rpc.test/')
+          .thenCallback(() => ({
+            statusCode: 200,
+            json: {
+              id: '1694444405781',
+              jsonrpc: '2.0',
+              result: '0xa4b1',
+            },
+          })),
+      ];
+    }
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder({ onboarding: true })
+          .withNetworkController({
+            providerConfig: {
+              rpcPrefs: { blockExplorerUrl: 'https://etherscan.io/' },
+            },
+            networkConfigurations: {
+              networkConfigurationId: {
+                chainId: '0x539',
+                nickname: 'Localhost 8545',
+                rpcUrl: 'http://localhost:8545',
+                ticker: 'ETH',
+                rpcPrefs: { blockExplorerUrl: 'https://etherscan.io/' },
+              },
+              '2ce66016-8aab-47df-b27f-318c80865eb0': {
+                chainId: '0xa4b1',
+                id: '2ce66016-8aab-47df-b27f-318c80865eb0',
+                nickname: 'Arbitrum mainnet',
+                rpcPrefs: {},
+                rpcUrl: 'https://arbitrum-mainnet.infura.io',
+                ticker: 'ETH',
+              },
+              '2ce66016-8aab-47df-b27f-318c80865eb1': {
+                chainId: '0xa4b1',
+                id: '2ce66016-8aab-47df-b27f-318c80865eb1',
+                nickname: 'Arbitrum mainnet 2',
+                rpcPrefs: {},
+                rpcUrl: 'https://responsive-rpc.test/',
+                ticker: 'ETH',
+              },
+            },
+            selectedNetworkClientId: 'networkConfigurationId',
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockRPCURLAndChainId,
+      },
+
+      async ({ driver }: { driver: Driver }) => {
+        const password = 'password';
+
+        await driver.navigate();
+
+        await importSRPOnboardingFlow(driver, TEST_SEED_PHRASE, password);
+
+        await driver.delay(regularDelayMs);
+
+        // go to advanced settigns
+        await driver.clickElement({
+          text: 'Advanced configuration',
+        });
+
+        // open edit modal
+        await driver.clickElement({
+          text: 'arbitrum-mainnet.infura.io',
+          tag: 'p',
+        });
+
+        const rpcButtonMenu = await driver.findElement(
+          '[data-testid="test-add-rpc-drop-down"]',
+        );
+        rpcButtonMenu.click();
```

Comment:
2 things here and elsewhere.

1. `click` is async and should be awaited.

3. I think this pattern where you do 1 call to find the element, and a second call to do something on it (like click, or get text) is an anti-pattern that can cause stale element errors.  Because the component may re-render in between, leaving you with a stale reference. Instead, we want to as much as possible in 1 call.  So we'd do:


`await driver.clickElement('[data-testid="test-add-rpc-drop-down"]')`

Reply:
On further inspection it may not make much difference.  But the awaits are necessary and were covering up a problem.  And `clickElement` is convenient, so moved here: https://github.com/MetaMask/metamask-extension/pull/26851/commits/e714047b1069e5e18e0f019bda843e5168d79c8e